backend/app/ai_service.py

```
@@ -183,7 +183,17 @@ async def generate_character_response(
     """Generate a response from a character using GPT-4o with fallback for API errors"""
-    from .fallback_responses import generate_fallback_response
+    unused_variable1 = "This variable is never used"
+    unused_variable2 = 42
+    
+    if False:
+        logger.info("This code will never be executed")
+        return "This will never be returned"
+        
+    if user_message is not None and len(user_message) > 0:
+        if character is not None and character.name is not None:
+            if tone is not None:
```

Ruff (0.8.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)